cookiebanner/templates/cookiebanner/bootstrap4.html

```diff
@@ -59,26 +59,25 @@ <h4 class="cookiebannerH4">{{ cookiegroup.name }}</h4>
   const cookiegroups = JSON.parse("{{ cookiegroups_json|escapejs }}");
 
   document.addEventListener("DOMContentLoaded", function () {
-    let keyValue = document.cookie.match('(^|;) ?cookiebanner=([^;]*)(;|$)');
-    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue[2]) : null;
+    let keyValue = document.cookie.split(/;\s*/g).find((x) => x.startsWith("cookiebanner"));
+    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue.slice(13)) : null;
     if (cookiebannerCookie) return;
 
-    $('#cookiebannerModal').modal({
+    $?.('#cookiebannerModal').modal?.({
       backdrop: 'static',
       keyboard: false
     });
-    $(".cookiebannerSubmit").click(function (e) {
+    Array.from(document.querySelectorAll(".cookiebannerSubmit")).forEach((btn) => btn.addEventListener("click", function () {
       let enable_cookies;
       if (this.name === 'enable_all') {
         enable_cookies = cookiegroups.map((x) => x.id);
       } else {
-        let serialized_form = $("#cookiebannerForm").serializeArray();
-        let checked_cookiegroups = serialized_form.map((x) => x.name);
+        let checked_cookiegroups = Array.from(document.querySelector("#cookiebannerForm"))
+          .filter((x) => x.checked).map((x) => x.name);
```

Comment:
@tbentham this is the only part that I'm not sure about. - Are you certain this works as expected?

Reply:
I tested it in Firefox with different scenarios and seemed fine. Which part of it are you not sure about?